### PR TITLE
Describe possible need for loading the 'lmodern' package when using PGF files

### PR DESCRIPTION
**Pull request by uihsnv**
## PR Summary

The PR adds a suggestion to use the `lmodern` package when using `.pgf` files generated by Matplotlib. Changes are made both to a documentation page as well as the commented preamble printed in every generated `.pgf` file.

This was found to be necessary specifically in the case of a math-bold-faced, uppercase Greek letter 'delta' was not rendered when the `lmodern` package was not loaded.

See discussion for more details:
https://matrix.to/#/!BXmyZMTnRjWJldDRLV:gitter.im/$M7KxLQuyUosx2byFxIsKJE8QS_14NRiWqmmlb6nSIfE?via=gitter.im&via=matrix.org

## PR Checklist

- [N/A] Has pytest style unit tests (and `pytest` passes).
- [N/A] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [N/A] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: tutorials/text/pgf.py [hunk 1]
```diff
@@ -187,6 +187,11 @@
   using either the ``rasterized=True`` keyword, or ``.set_rasterized(True)`` as
   per :doc:`this example </gallery/misc/rasterization_demo>`.
 
+* Various math font are compiled and rendered only if corresponding font
+  packages are loaded. The ``lmodern`` package is often required when using
+  math symbols. See `discussion <https://matrix.to/#/!BXmyZMTnRjWJldDRLV:gitter.im/$M7KxLQuyUosx2byFxIsKJE8QS_14NRiWqmmlb6nSIfE?via=gitter.im&via=matrix.org>`
```
Comment (jklymak) [inline tutorials/text/pgf.py:192]: I don't think linking a gitter discussion (via matrix!) is something we should start doing in our docs.

Comment (tacaswell) [reply tutorials/text/pgf.py:192]: I agree, could you summarize the discussion here?  

We should try to, with in reason, keep our docs self-contained and usable by people in off-line situations (for example on an airplane or boat).

**Comment by anntzer** [comment]
I would have thought(?) that this is rather a standard TeX limitation (https://texfaq.org/FAQ-boldgreek, https://tex.stackexchange.com/questions/595) and may not warrant a specific doc entry on our side?

**Comment by uihsnv** [comment]
@anntzer Those links describe a related issue of `\mathbf{}` not sufficiently emboldening some Greek font. But here, the very font was missing when compiling without the `lmodern` package.

File: tutorials/text/pgf.py (hunk 1 above)
Comment (uihsnv) [reply tutorials/text/pgf.py]: I've made the discussion more specific.

**Comment by jklymak** [comment]
I think the mystery here is what combination of factors do you need to trigger this problem.  I've never come across a LaTeX install that requires `\include{lmodern}`for most typesetting needs.

**Comment by tacaswell** [comment]
On one hand, that we have this issue is super weird (and as discussed on gitter it seems to depend  on which LaTeX engine you use) and may actually be caused by something inside of the pgf package.

Given that this work around does fix the problem, I am in favor of including a note about this and opening a new issue in the spirit of "sort out when / why / if `lmodern` is actually needed".
